modules/twinklearv.js

```diff
@@ -18,14 +18,57 @@ Twinkle.arv = function twinklearv() {
 		return;
 	}
 
-	var title = mw.util.isIPAddress(username) ? 'Report IP to administrators' : 'Report user to administrators';
+	var isIP = mw.util.isIPAddress(username);
+	var title = isIP ? 'Report IP to administrators' : 'Report user to administrators';
 
 	Twinkle.addPortletLink(function() {
-		Twinkle.arv.callback(username);
+		Twinkle.arv.callback(username, isIP);
 	}, 'ARV', 'tw-arv', title);
 };
 
-Twinkle.arv.callback = function (uid) {
+Twinkle.arv.callback = function (username, isIP) {
+	var params = {
+		format: 'json',
+		action: 'query',
+		list: 'blocks',
+		bkprop: 'range'
+	};
+	if (isIP) {
+		params.bkip = username;
+	} else {
+		params.bkusers = username;
+	}
+
+	new mw.Api().get(params).then(function(data) {
```

Comment:
Presumably this would be nicer after #765 ?

Reply:
Not sure - this is based on code in checking for blocks elsewhere in the repo (don't remember where exactly) but morebits api may not be needed

Reply:
[twinkleblock](https://github.com/azatoth/twinkle/blob/f1135f6304300eb55624e38ece6e921e209afc4b/modules/twinkleblock.js#L98-L107).  No, it's not needed, but I believe it's the kind of thing that #765 would enable `Morebits.wiki.api` to handle (note: I haven't really reviewed #765).  Not necessarily suggesting you change it, just noting its existence, especially given that it's an odd-one-out.